src/core/auth/wallet-authenticator.ts

```diff
@@ -0,0 +1,184 @@
+import { ethers } from "ethers";
+import { SDKOptions } from "../../schema";
+import {
+  AuthenticatedPayload,
+  AuthenticatedPayloadSchema,
+  AuthenticationPayloadInput,
+  AuthenticationPayloadInputSchema,
+  AuthenticationPayloadOutput,
+  AuthenticationPayloadOutputSchema,
+  AuthorizedPayload,
+  AuthorizedPayloadSchema,
+  FilledAuthenticationPayloadInput,
+} from "../../schema/auth";
+import { RPCConnectionHandler } from "../classes/rpc-connection-handler";
+import { NetworkOrSignerOrProvider } from "../types";
+
+/**
+ * Wallet Authenticator
+ * @remarks Enables sign-in with ethereum based authentication.
```

Comment:
could expand a bit more on the fact that this is using a JWT style authentication system, token can be re-used for multiple calls, and does not need to be stored anywhere

Reply:
It would need to be stored on the client-side if they want to be able to make multiple requests w/o signing in again - but the nice thing is doesn't need any storage in a database